src/api/agencyList.ts

```diff
@@ -1,33 +1,21 @@
-import { useEffect, useState } from 'react'
-import { STRIDE_API_BASE_PATH } from './apiConfig'
-import dayjs from 'src/dayjs'
+import { GtfsAgencyPydanticModel } from '@hasadna/open-bus-api-client'
+import { GTFS_API } from './apiConfig'
 
-export interface Agency {
-  date: string // example - "2019-07-01"
-  operator_ref: number // example - 25,
-  agency_name: string // example - "אלקטרה אפיקים"
-}
-
-let agencyList: Agency[]
+let agencyList: GtfsAgencyPydanticModel[]
 
 const tryDates = [
-  dayjs().subtract(1, 'day').format('YYYY-MM-DD'),
-  dayjs().subtract(8, 'day').format('YYYY-MM-DD'),
-  '2025-05-18',
+  new Date(Date.now() - 24 * 60 * 60 * 1000),
+  new Date(Date.now() - 7 * 24 * 60 * 60 * 1000),
+  new Date('2025-05-18'),
 ]
 
-/**
- * Fetch agency data from MOT api
- * @returns Agency data array, might contain DUPLICATE agencies with different `date` values
- */
-export default async function getAgencyList(): Promise<Agency[]> {
+export default async function getAgencyList(): Promise<GtfsAgencyPydanticModel[]> {
   if (!agencyList) {
-    let data: Agency[] = []
+    let data: GtfsAgencyPydanticModel[] = []
```

Comment:
if data will store a promise for the data instead of the actual data, it will prevent a race conditions where `getAgencyList` is called twice and two requests are executed